server/src/main/scala/org/scassandra/server/cqlmessages/CqlProtocolHelper.scala

```diff
@@ -18,17 +18,37 @@ package org.scassandra.server.cqlmessages
 import java.net.InetAddress
 import java.util.UUID
 
-import akka.util.{ByteIterator, ByteString}
-import io.netty.buffer.{ByteBuf, Unpooled}
-import org.apache.cassandra.transport.CBUtil
+import akka.util.{ByteIterator, ByteStringBuilder, ByteString}
+
 import org.scassandra.server.cqlmessages.types.ColumnType
 
 import scala.collection.JavaConverters._
+import scala.collection.{mutable, SeqLike}
 
 object CqlProtocolHelper {
   implicit val byteOrder = java.nio.ByteOrder.BIG_ENDIAN
   val NullValue: Array[Byte] = Array[Byte](-1, -1, -1, -1)
 
+  def withLength(f: (ByteStringBuilder, Int) => ByteStringBuilder)(bytes: Array[Byte]) = {
+    val bs = f(ByteString.newBuilder, bytes.length)
+    bs.putBytes(bytes)
+    bs.result().toArray
+  }
+
+  def withIntLength = withLength(_.putInt(_)) _
+
+  def withShortLength = withLength(_.putShort(_)) _
+
+  def elementSerializer(value: Any, f: (Any) => Array[Byte])(implicit protocolVersion: ProtocolVersion): Array[Byte] = {
+    val bytes = f(value)
+    protocolVersion match {
+      // proto <= v2 uses short length for elements, >= 3 uses int length.
+      case VersionOne | VersionTwo => withShortLength(bytes)
```

Comment:
I do this in a number of places and it should probably be generalized (will work on that).